src/ripple/protocol/impl/STArray.cpp

```diff
@@ -144,16 +144,12 @@ std::string STArray::getText () const
 Json::Value STArray::getJson (int p) const
 {
     Json::Value v = Json::arrayValue;
-    int index = 1;
     for (auto const& object: v_)
     {
         if (object.getSType () != STI_NOTPRESENT)
         {
             Json::Value& inner = v.append (Json::objectValue);
-            auto const& fname = object.getFName ();
-            auto k = fname.hasName () ? fname.fieldName : std::to_string(index);
-            inner[k] = object.getJson (p);
-            index++;
+            inner [object.getFName().getJsonName()] = object.getJson (p);
```

Comment:
`jasonName` can be empty, so we might overwrite previous entries here.

Reply:
Only for outer containers like generic and invalid. You can't have a generic or an invalid inside an object.